### PR TITLE
Alerting: Validate that tags are 100 characters or less

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**What is this feature?**

This pull request fixes a bug where tags longer than 100 characters would fail due to a column constraint of 100 characters. Instead, a validation error is returned and shown in the UI:

Screenshot 2023-01-27 at 15 04 06

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/services/backend_srv.ts
Comment (gillesdemey) [inline public/app/core/services/backend_srv.ts:320]: I feel like there _has_ to be here for a reason for this check to have existed 👀

**Review by santihernandezc — COMMENTED** [review]
The BE changes look fine, but I'll let someone from FE approve this PR. We should confirm that the changes in the UI are ok.

File: public/app/core/services/backend_srv.ts
Comment (grobinson-grafana) [reply public/app/core/services/backend_srv.ts:320]: I think its broken so I removed it. If the message is `< 80` then the error that's shown in the UI is just `Validation failed`. You won't see the actual error from the API.

**Comment by heldersepu** [comment]
This PR fixes part of #60851 
It would be nice to also increase the 100  limit to 400

**Comment by armandgrillet** [comment]
@heldersepu Legacy alerting is deprecated thus we limit changes to bug fixes. A limit increase would not be a bug fix and would impact dashboard tags which is why it will not be done in this PR.

File: public/app/core/services/backend_srv.ts
Comment (gillesdemey) [reply public/app/core/services/backend_srv.ts:320]: I'm pretty sure what's happening here before is to move the error to the "description" of an alert if it's considered "too long" and show "Error" as the title.

In this screenshot you can see an alert with "title and description" and how it differs from "just a title".

image

I'm not super comfortable merging these changes - can we pair up to figure out why it doesn't seem to be working for you without the changes proposed in this PR?

**Review by gillesdemey — CHANGES_REQUESTED** [review]
Let's figure out why `if (message.length > 80)` was there before we proceed with merging :)

File: public/app/core/services/backend_srv.ts
Comment (grobinson-grafana) [reply public/app/core/services/backend_srv.ts:320]: The issue is that if we keep the following conditional:

```js
if (message.length > 80) {
      description = message;
      message = 'Error';
}
```

Then the error message is never shown to the user because it is shorter than 80 characters:

Screenshot 2023-01-31 at 10 32 27

Comment (gillesdemey) [reply public/app/core/services/backend_srv.ts:320]: That seems like either a regression of the alert component or this has never worked at all 🤔

Comment (grobinson-grafana) [reply public/app/core/services/backend_srv.ts:320]: I would guess it has never worked at all, we just didn't know as the API never returned an error message shorter than 80 characters for HTTP 422

**Comment by grobinson-grafana** [comment]
> Let's figure out why `if (message.length > 80)` was there before we proceed with merging :)

I really think it's a mistake, the original code was:

```js
// Validation
if (err.status === 422) {
  message = 'Validation failed';
}
```

when it should have been:

```js
// Validation
if (err.status === 422) {
  description = message;
  message = 'Validation failed';
}
```

File: public/app/core/services/backend_srv.ts
Comment (gillesdemey) [reply public/app/core/services/backend_srv.ts:320]: Oh I see what's going on here, I think there's a better way to solve this for `422` status code errors without impacting the existing logic for the other status codes.

**Review by gillesdemey — APPROVED** [review]

**Comment by anuragarora** [comment]
The fix failed to be backported to v8.4.x and v8.5.x. Can this change be backported manually ?

**Pull request merged**
